docs/manuals/core/concepts/index.md

```diff
@@ -33,27 +33,33 @@ pipelines.
 
         ![Linear pipeline](../pic/branching_algo.svg)
 
-The purpose of this section is to define the following Taipy Core concepts.
-
-- A [Data node](data-node.md) (the dark blue boxes) represents a dataset. It can be shared by multiple tasks as input or
-  output.
-- A [Task](task.md) (the orange boxes) can be seen as a function taking some data node(s) as input and returning
-  some data node(s).
-- A [Job](job.md) is a unique execution of a Task.
-- A [Pipeline](pipeline.md) represents an algorithm or pipeline made of tasks that should run together.
+This section aims at defining the following Taipy Core concepts.
+
+- A [Data node](data-node.md) (the dark blue boxes) represents a reference to a dataset. It can refer to any type of
+  data such as built-in Python objects (e.g. integer, string, dictionary of parameters, list of values, ...) or more
+  complex objects (e.g. a machine learning model, a list of custom objects, the result of a database query, ...)
+  A data node can be shared by multiple tasks as input or output.
+- A [Task](task.md) (the orange boxes) can be seen as a function receiving some data node(s) as input and returning
+  some data node(s) as output.
+- A [Job](job.md) represents a unique execution of a Task.
+- A [Pipeline](pipeline.md) represents a set of tasks connected together through data nodes, that should be executed
+  together and forming a consistent algorithm.
```

Comment:
'together' may imply 'simultaneously'. May be 'as a whole'?

Reply:
Indeed, that could remove the possible confusion. Thx!